File: lib/DevOther/src/main/java/java/dev/engine/barcode/BarCodeData.java
package java.dev.engine.barcode;

import android.graphics.Bitmap;
import android.graphics.Color;

import com.google.zxing.BarcodeFormat;

import dev.engine.barcode.IBarCodeEngine;

/**
 * detail: BarCode ( Data、Params ) Item
 * @author Ttt
 */
public class BarCodeData
        extends IBarCodeEngine.EngineItem {

    // 条码内容
    private final String        mContent;
    // 条码宽度
    private final int           mWidth;
    // 条码高度
    private final int           mHeight;
    // 条码类型 ( 默认二维码 )
    private       BarcodeFormat mFormat          = BarcodeFormat.QR_CODE;
    // 条码前景色 ( 默认黑色方块 )
    private       int           mForegroundColor = Color.BLACK;
    // 条码背景色 ( 默认白色背景 )
    private       int           mBackgroundColor = Color.WHITE;
    // 条码嵌入 icon、logo
    private       Bitmap        mIcon;

    private BarCodeData(
            String content,
            int width,
            int height
    ) {
        this.mContent = content;
        this.mWidth   = width;
        this.mHeight  = height;
    }

    // ==========
    // = 快捷方法 =
    // ==========

    /**
     * 快捷创建 BarCode ( Data、Params ) Item
     * @param content 条码内容
     * @param size    条码大小
     * @return BarCode Item
     */
    public static BarCodeData get(
            String content,
            int size
    ) {
        return new BarCodeData(content, size, size);
    }

    /**
     * 快捷创建 BarCode ( Data、Params ) Item
     * @param content 条码内容
     * @param width   条码宽度
     * @param height  条码高度
     * @return BarCode Item
     */
    public static BarCodeData get(
            String content,
            int width,
            int height
    ) {
        return new BarCodeData(content, width, height);
    }

    // ===========
    // = get/set =
    // ===========

    /**
     * 获取条码内容
     * @return 条码内容
     */
    public String getContent() {
        return mContent;
    }

    /**
     * 获取条码宽度
     * @return 条码宽度
     */
    public int getWidth() {
        return mWidth;
    }

    /**
     * 获取条码高度
     * @return 条码高度
     */
    public int getHeight() {
        return mHeight;
    }

    /**
     * 获取条码类型
     * @return 条码类型
     */
    public BarcodeFormat getFormat() {
        return mFormat;
    }

    /**
     * 设置条码类型
     * @param format 条码类型
     * @return BarCode Item
     */
    public BarCodeData setFormat(BarcodeFormat format) {
        this.mFormat = format;
        return this;
    }

    /**
     * 获取条码前景色
     * @return 条码前景色
     */
    public int getForegroundColor() {
        return mForegroundColor;
    }

    /**
     * 设置条码前景色
     * @param foregroundColor 条码前景色
     * @return BarCode Item
     */
    public BarCodeData setForegroundColor(int foregroundColor) {
        this.mForegroundColor = foregroundColor;
        return this;
    }

    /**
     * 获取条码背景色
     * @return 条码背景色
     */
    public int getBackgroundColor() {
        return mBackgroundColor;
    }

    /**
     * 设置条码背景色
     * @param backgroundColor 条码背景色
     * @return BarCode Item
     */
    public BarCodeData setBackgroundColor(int backgroundColor) {
        this.mBackgroundColor = backgroundColor;
        return this;
    }

    /**
     * 设置条码前景、背景色
     * @param foregroundColor 条码前景色
     * @param backgroundColor 条码背景色
     * @return BarCode Item
     */
    public BarCodeData setBackgroundColor(
            int foregroundColor,
            int backgroundColor
    ) {
        this.mForegroundColor = foregroundColor;
        this.mBackgroundColor = backgroundColor;
        return this;
    }

    /**
     * 获取条码嵌入 icon、logo
     * @return 嵌入 icon、logo
     */
    public Bitmap getIcon() {
        return mIcon;
    }

    /**
     * 设置条码嵌入 icon、logo
     * @param icon icon、logo
     * @return BarCode Item
     */
    public BarCodeData setIcon(Bitmap icon) {
        this.mIcon = icon;
        return this;
    }
}

File: lib/DevEngine/src/main/java/dev/engine/core/barcode/BarCodeData.kt
package dev.engine.core.barcode

import android.graphics.Bitmap
import android.graphics.Color
import com.google.zxing.BarcodeFormat
import dev.engine.barcode.IBarCodeEngine

/**
 * detail: BarCode ( Data、Params ) Item
 * @author Ttt
 */
open class BarCodeData private constructor(
    // 条码内容
    private val mContent: String?,
    // 条码宽度
    private val mWidth: Int,
    // 条码高度
    private val mHeight: Int
) : IBarCodeEngine.EngineItem {

    companion object {

        /**
         * 快捷创建 BarCode ( Data、Params ) Item
         * @param content 条码内容
         * @param size    条码大小
         * @return BarCode Item
         */
        operator fun get(
            content: String,
            size: Int
        ): BarCodeData {
            return BarCodeData(content, size, size)
        }

        /**
         * 快捷创建 BarCode ( Data、Params ) Item
         * @param content 条码内容
         * @param width   条码宽度
         * @param height  条码高度
         * @return BarCode Item
         */
        operator fun get(
            content: String,
            width: Int,
            height: Int
        ): BarCodeData {
            return BarCodeData(content, width, height)
        }
    }

    // 条码类型 ( 默认二维码 )
    private var mFormat = BarcodeFormat.QR_CODE

    // 条码前景色 ( 默认黑色方块 )
    private var mForegroundColor = Color.BLACK

    // 条码背景色 ( 默认白色背景 )
    private var mBackgroundColor = Color.WHITE

    // 条码嵌入 icon、logo
    private var mIcon: Bitmap? = null

    // icon 占比
    private var mIconScale = 4.0F

    // ===========
    // = get/set =
    // ===========

    /**
     * 获取条码内容
     * @return 条码内容
     */
    fun getContent(): String? {
        return mContent
    }

    /**
     * 获取条码宽度
     * @return 条码宽度
     */
    fun getWidth(): Int {
        return mWidth
    }

    /**
     * 获取条码高度
     * @return 条码高度
     */
    fun getHeight(): Int {
        return mHeight
    }

    /**
     * 获取条码类型
     * @return 条码类型
     */
    fun getFormat(): BarcodeFormat {
        return mFormat
    }

    /**
     * 设置条码类型
     * @param format 条码类型
     * @return BarCode Item
     */
    fun setFormat(format: BarcodeFormat): BarCodeData {
        mFormat = format
        return this
    }

    /**
     * 获取条码前景色
     * @return 条码前景色
     */
    fun getForegroundColor(): Int {
        return mForegroundColor
    }

    /**
     * 设置条码前景色
     * @param foregroundColor 条码前景色
     * @return BarCode Item
     */
    fun setForegroundColor(foregroundColor: Int): BarCodeData {
        mForegroundColor = foregroundColor
        return this
    }

    /**
     * 获取条码背景色
     * @return 条码背景色
     */
    fun getBackgroundColor(): Int {
        return mBackgroundColor
    }

    /**
     * 设置条码背景色
     * @param backgroundColor 条码背景色
     * @return BarCode Item
     */
    fun setBackgroundColor(backgroundColor: Int): BarCodeData {
        mBackgroundColor = backgroundColor
        return this
    }

    /**
     * 设置条码前景、背景色
     * @param foregroundColor 条码前景色
     * @param backgroundColor 条码背景色
     * @return BarCode Item
     */
    fun setBackgroundColor(
        foregroundColor: Int,
        backgroundColor: Int
    ): BarCodeData {
        mForegroundColor = foregroundColor
        mBackgroundColor = backgroundColor
        return this
    }

    /**
     * 获取条码嵌入 icon、logo
     * @return 嵌入 icon、logo
     */
    fun getIcon(): Bitmap? {
        return mIcon
    }

    /**
     * 设置条码嵌入 icon、logo
     * @param icon icon、logov
     * @return BarCode Item
     */
    fun setIcon(icon: Bitmap?): BarCodeData {
        mIcon = icon
        return this
    }

    /**
     * 获取 icon 占比
     * @return icon 占比
     */
    fun getIconScale(): Float {
        return mIconScale
    }

    /**
     * 设置 icon 占比
     * @param iconScale icon 占比
     * @return BarCode Item
     */
    fun setIconScale(iconScale: Float): BarCodeData {
        mIconScale = iconScale
        return this
    }
}